dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java

```diff
@@ -995,7 +1000,42 @@ protected List<Contentlet> findAllVersions(final Identifier identifier, final bo
         return findAllVersions(identifier, bringOldVersions, null);
     }
 
-	@Override
+    @Override
+    protected  List<Contentlet> findAllVersions(final Identifier identifier, final Variant variant)
+            throws DotDataException, DotSecurityException{
+        DotPreconditions.notNull(identifier, () -> "Identifier cannot be null");
+        DotPreconditions.notNull(variant, () -> "Variant cannot be null");
+
+        final String columnName = getJsonVariantIdColumnName();
+
+        final List<Map<String,Object>> list = new DotConnect()
+                .setSQL(String.format("select inode from contentlet where identifier = ? and %s = ?",
+                        columnName))
+                .addParam(identifier.getId())
+                .addParam(variant.name())
+                .loadResults();
+
+        return findContentlets(list.stream()
+                .map(map -> map.get("inode").toString())
+                .collect(Collectors.toList()));
+    }
+
+    @Nullable
+    private static String getJsonVariantIdColumnName() {
+        String columnName = null;
```

Comment:
What happens if the current data set is from an old customer? I think this method will return null instead of the legacy expected column, we need to check that scenario.